drivers/sensor/vl53l0x/vl53l0x.c

```diff
@@ -0,0 +1,250 @@
+#include <sensor.h>
+#include <init.h>
+#include <gpio.h>
+#include <misc/__assert.h>
+
+#include "vl53l0x.h"
+
+static int vl53l0x_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct vl53l0x_data *drv_data = (struct vl53l0x_data *)dev->driver_data;
+	u8_t ret = 0;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL
+			|| chan == SENSOR_CHAN_DISTANCE
+			|| chan == SENSOR_CHAN_PROX);
+
+	ret = VL53L0X_PerformSingleRangingMeasurement(&drv_data->vl53l0x,
+					&drv_data->RangingMeasurementData);
+	if (ret < 0) {
+		SYS_LOG_ERR("Could not perform measurment (error=%d)", ret);
+		return -EINVAL;
+	}
+
+	return ret;
+}
+
+
+static int vl53l0x_channel_get(struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct vl53l0x_data *drv_data = (struct vl53l0x_data *)dev->driver_data;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_DISTANCE
+			|| chan == SENSOR_CHAN_PROX);
+
+	val->val1 = drv_data->RangingMeasurementData.RangeMilliMeter;
+	val->val2 = 0;
+
+	if (chan == SENSOR_CHAN_PROX) {
+		if (val->val1 <= CONFIG_VL53L0X_PROXIMITY_THRESHOLD) {
+			val->val1 = 1;
+		} else {
+			val->val1 = 0;
+		}
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api vl53l0x_api_funcs = {
+	.sample_fetch = vl53l0x_sample_fetch,
+	.channel_get = vl53l0x_channel_get,
+};
+
+static int vl53l0x_setup_single_shot(struct device *dev)
+{
+	int ret;
+	u8_t VhvSettings;
+	u8_t PhaseCal;
+	u32_t refSpadCount;
+	u8_t isApertureSpads;
+	struct vl53l0x_data *drv_data = dev->driver_data;
+
+	ret = VL53L0X_StaticInit(&drv_data->vl53l0x);
+
+	ret = VL53L0X_PerformRefCalibration(&drv_data->vl53l0x,
+					    &VhvSettings,
+					    &PhaseCal);
+	if (ret) {
+		SYS_LOG_DBG("VL53L0X_PerformRefCalibration failed");
+		goto exit;
+	}
+
+	ret = VL53L0X_PerformRefSpadManagement(&drv_data->vl53l0x,
+					       &refSpadCount,
+					       &isApertureSpads);
+	if (ret) {
+		SYS_LOG_ERR("VL53L0X_PerformRefSpadManagement failed");
+		goto exit;
+	}
+
+	ret = VL53L0X_SetDeviceMode(&drv_data->vl53l0x,
+				    VL53L0X_DEVICEMODE_SINGLE_RANGING);
+	if (ret) {
+		SYS_LOG_ERR("VL53L0X_SetDeviceMode failed");
+		goto exit;
+	}
+
+	ret = VL53L0X_SetLimitCheckEnable(&drv_data->vl53l0x,
+					  VL53L0X_CHECKENABLE_SIGMA_FINAL_RANGE,
+					  1);
+	if (ret) {
+		SYS_LOG_ERR("VL53L0X_SetLimitCheckEnable failed");
+		goto exit;
+	}
+
+	ret = VL53L0X_SetLimitCheckEnable(&drv_data->vl53l0x,
+				VL53L0X_CHECKENABLE_SIGNAL_RATE_FINAL_RANGE,
+				1);
+	if (ret) {
+		SYS_LOG_ERR("VL53L0X_SetLimitCheckEnable failed");
+		goto exit;
+	}
+
+	ret = VL53L0X_SetLimitCheckValue(&drv_data->vl53l0x,
+				VL53L0X_CHECKENABLE_SIGNAL_RATE_FINAL_RANGE,
+				VL53L0X_SETUP_SIGNAL_LIMIT);
+
+	if (ret) {
+		SYS_LOG_ERR("VL53L0X_SetLimitCheckValue failed");
+		goto exit;
+	}
+
+	ret = VL53L0X_SetLimitCheckValue(&drv_data->vl53l0x,
+					 VL53L0X_CHECKENABLE_SIGMA_FINAL_RANGE,
+					 VL53L0X_SETUP_SIGMA_LIMIT);
+	if (ret) {
+		SYS_LOG_ERR("VL53L0X_SetLimitCheckValue failed");
```

Comment:
Same error message as upper.
Maybe we can use:
 "VL53L0X_SetLimitCheckValue signal rate failed" and  "VL53L0X_SetLimitCheckValue sigma failed"